### PR TITLE
std.traits now has hasAttribute to simplify checking for the presence…

### PR DESCRIPTION
**Pull request by atilaneves**
… of UDAs

### CODE REVIEW

File: profilegc.log [hunk 1]
```diff
@@ -0,0 +1 @@
+bytes allocated, type, function, file:line
```
Comment (MetaLang) [inline profilegc.log]: ???

**Comment by MetaLang** [comment]
I was annoyed that the following didn't work, as it seems like it should just from the name of hasAttribute:

```d
@("test")
void test() {}

void main()
{
	assert(hasAttribute!(test, "test"));
}
```

Perhaps a better name would be `hasAttributeOfType`.

File: profilegc.log (hunk 1 above)
Comment (atilaneves) [inline profilegc.log]: Oops.

**Comment by atilaneves** [comment]
I changed the name. The issue here is that I _could_ have written a `hasAttribute` that accepted types or values, and it's not that hard. The issue is passing in a primitive type, because weirdly enough you can't pass an `int` as a template alias parameter.
The other alternative is a `hasAttribute` that takes parameters that are types or values, _unless_ they're primitive types, and a `hasPrimitiveAttribute` for those.

**Comment by MetaLang** [comment]
>The issue is passing in a primitive type...

```d
template test(alias i)
{
	enum test = i == 0;
}

void main()
{
	assert(test!0);
}
```

**Comment by jcrapuchettes** [comment]
This is similar to #3356

**Comment by atilaneves** [comment]
@MetaLang In your example, `test!0` works, but `test!int` doesn't.

**Comment by atilaneves** [comment]
@jcrapuchettes It's nearly identical. This is what happens when we talk at DConf then the both of us submit pull requests. I think the other one is better actually, I'm closing this one.

**Comment by MetaLang** [comment]
Ah, I see what you mean. You need to do the following:

```d
template test(T...)
if (T.length == 1)
{
    //T can be anything in here
}
```

**Pull request closed without merging**